src/main/java/bio/terra/service/dataset/flight/datadelete/DatasetDataDeleteFlight.java

```diff
@@ -64,6 +68,23 @@ public DatasetDataDeleteFlight(FlightMap inputParameters, Object applicationCont
     addStep(
         new LockDatasetStep(datasetService, UUID.fromString(datasetId), true), lockDatasetRetry);
 
+    // Note: don't worry about Azure until we tackle DR-2252
+    boolean autocommit;
+    if (request.getTransactionId() == null) {
+      // Note: don't unlock transaction so we keep a history of what flight an auto-commit
+      // transaction was created for
+      String transactionDesc = "Autocommit transaction";
+      addStep(
+          new TransactionOpenStep(
+              datasetService, bigQueryPdao, userReq, transactionDesc, false, false));
+      autocommit = true;
+    } else {
+      addStep(
+          new TransactionLockStep(
+              datasetService, bigQueryPdao, request.getTransactionId(), true, userReq));
```

Comment:
Wouldn't `request.getTransactionId()` be null in this case?

Reply:
I don't think so.  We check for null in the if branch and then the else branch is explicitly when it's not null

Reply:
Oh duh, I have it backwards.